### ISSUE TITLE
current_page variable not exported everytime

### ISSUE THREAD

**Issue by gsouf** [mkdocs/mkdocs#875]
Hi,

Opening this issue as a follow for #266

The ``current_page`` variable is unavailable for templates that are not built from a ``.md`` file. For instance ``404.html``

How to reproduce:

```
mkdocs new test-project
cd test-project/
mkdir theme
echo "theme_dir: theme" >> mkdocs.yml
echo "aaa {{ current_page.something }}" > theme/base.html
mkdocs build
```

At this point it works.

Now do:

```
echo "{% extends 'base.html' %}" > theme/404.html
mkdocs build
```

And that will fail

**Comment by gsouf** [comment]
It appears to be working when adding the ``current_page`` config in this array: 
https://github.com/mkdocs/mkdocs/blob/0.15.3/mkdocs/commands/build.py#L58

**Comment by gsouf** [comment]
Same happens with the ``meta`` variable

**Comment by waylan** [comment]
@gsouf thanks for the clear report. This is definitely a different problem from the one discussed in #266 as this only applies to non-Markdown files. As I understand it, only Markdown files get a "page context" when passed to the template, whereas every template gets the "global context". So what you are requesting is a new feature: that a "page context" be created for non-Markdown files. Up until recently it wasn't even possible to use non-Markdown files with templates, so I'm not sure if this exclusion was intentional or not. We'll have to wait for @d0ugal to respond (I understand he is unavailable for a few more days).

**Comment by gsouf** [comment]
@waylan you got the issue. 

I dont know much about the internal of mkdocs, but I think that setting an empty array is a good fix, because when a non-markdown file extends a base template that uses it, jinja will throw an error. 

Let's way for @d0ugal opinion!

**Comment by gsouf** [comment]
ping @d0ugal

**Comment by d0ugal** [comment]
I am currently in Yosemite on vacation. So I've not considered this fully, however, from a quick read the exclusion for non-markdown files wasn't intentional.

**Comment by gsouf** [comment]
Enjoy ;)

**Comment by frob** [comment]
I am getting this same behavior with custom themes on markdown files.

**Comment by frob** [comment]
With me the odd thing about it is, I can use current_page so long as I am testing for it first.

_Works:_

```
{% if current_page %} {{ current_page.input_path }} {% endif %}
```

_Doesn't Work:_

```
{{ current_page.input_path }}
```

This is using the latest pip installed 0.15.3

**Comment by waylan** [comment]
@frob is correct. By default the template system will not fail on an undeclared variable if all you do is check for its existence. It will return `False` instead, which makes for a nice safety net.  That said, it does not work as reliably as I would like, so in #921 I changed it so that the `page` variable will exist for every page, but equate to `None` when the page is not a Markdown page. Therefore, you will still need to check for its existence in the same manner:

    {% if page %} {{ page.input_path }} {% endif %}

You will want to review the documentation and release notes for 0.16 when it is released, or as part of the changes in #921.

**Comment by waylan** [comment]
Closing this as it was fixed in #921.
